fe/fe-core/src/main/java/org/apache/doris/statistics/util/StatisticsUtil.java

```diff
@@ -974,4 +976,28 @@ public static boolean isMvColumn(TableIf table, String columnName) {
             || columnName.startsWith(CreateMaterializedViewStmt.MATERIALIZED_VIEW_AGGREGATE_NAME_PREFIX);
     }
 
+    public static boolean isEmptyTable(TableIf table, AnalysisInfo.AnalysisMethod method) {
+        int waitRowCountReportedTime = 90;
+        if (!(table instanceof OlapTable) || method.equals(AnalysisInfo.AnalysisMethod.FULL)) {
+            return false;
+        }
+        OlapTable olapTable = (OlapTable) table;
+        for (int i = 0; i < waitRowCountReportedTime; i++) {
+            if (olapTable.getRowCount() > 0) {
+                return false;
+            }
+            // If visible version is 2, table is probably not empty. So we wait row count to be reported.
+            // If visible version is not 2 and getRowCount return 0, we assume it is an empty table.
+            if (olapTable.getVisibleVersion() != TableAttributes.TABLE_INIT_VERSION + 1) {
+                return true;
+            }
+            try {
+                Thread.sleep(1000);
```

Comment:
Will this waiting hold up the Worker thread? AFAIK, number of threads is limited in the analysisTaskExecutor.

Reply:
Yes, it will hold up the worker thread. But I don't think it's a problem. Because in most case, we don't need to wait, only when the visible version is 2 and get row count == 0, and the waiting time is usually less than 1 minute. Also, couple of minutes delay for auto analyze is acceptable.

Reply:
A delay for auto analysis is acceptable, but what if it is a manual one with sync? Queuing in the task queue will make users wait longer.

Reply:
Each manual analyze with sync operation will create its own thread pool, so it will not block other session.